### ISSUE TITLE
Zoom factor for spreadsheets seems off on macOS Big Sur

### ISSUE THREAD

**Issue by accolon** [CollaboraOnline/online#654]
**Describe the bug**
I'm using CODE with Nextcloud. I just updated my MacBook Pro to macOS Big Sur, and now the zoom factor in spreadsheets seems to be off. At the standard 100%, the contents are barely readable, switching to 200% looks like the old default. See attached screenshots and compare the table contents with the menu bar. At first I supposed this was a problem with the new Safari 14 that is bundled with Big Sur, but the same problem exists in Firefox.

This might have to do with the screen resolution scaling on MacBooks. My display has a native resolution of 3072 × 1920 but is scaled to "look like 1792 x 1120" in the default setting for this device. There are other applications (like VMware Fusion) which sometimes have scaling problems that look quite similar.

**To Reproduce**
1. Use MacBook with Retina display and screen scaling (in my case: MBP 16" 2019) with macOS Big Sur
2. Open or create a new ODS file with Collabora in Nextcloud
3. Leave zoom level at 100%

**Expected behavior**
At 100%, the table contents are clearly readable and correctly sized in comparison to the Collabora menu bar and other control elements.

**Actual behavior**
At 100%, the table contents are very small and almost illegible while the menu bar and other control elements look fine.

**Screenshots**
See attached, 100% and 200% zoom level of a newly created ODS file.

**Desktop (please complete the following information)**
 - OS: macOS Big Sur 11.0.1
 - Browser: Safari 14.0.1, Firefox 82.0.3
 - Device: MacBook Pro 16" 2019

**Additional context**
CODE 6.4-10 with richdocuments 3.7.9 on Nextcloud 20.0.1
<img width="891" alt="collabora_1x" src="https://user-images.githubusercontent.com/126886/99234292-a47b9c80-27f4-11eb-89df-b9d9cb9fe421.png">
<img width="891" alt="collabora_2x" src="https://user-images.githubusercontent.com/126886/99234303-a6ddf680-27f4-11eb-8573-09fdfe4ff001.png">

**Comment by thebearon** [comment]
I don't have a Mac with Big Sur, but #740 seems to be about the same issue, and confirming this.

When you write:
> My display has a native resolution of 3072 × 1920 but is scaled to "look like 1792 x 1120" in the default setting for this device.

Can you share where and what the exact setting is for that?

**Comment by accolon** [comment]
The setting is in "System Preferences" / "Displays".

I tried changing the scaled resolution, but this does only increase or decrease the size of the control elements, the table cells stay the same.

<img width="780" alt="Screenshot 2020-11-23 at 08 09 51" src="https://user-images.githubusercontent.com/126886/99936501-946b3c00-2d63-11eb-92ea-9e4a70adeb4b.png">

**Comment by tml1024** [comment]
Can reproduce on a 5K iMac running Big Sur. (Using #Resolution: Default for display.")

**Comment by tml1024** [comment]
Attempting to bisect. Although it is entirely possible that when using Safari on macOS 11, none of the previous states will be good.

**Comment by tml1024** [comment]
Bisecting points to d80b75ae9a9d2d56a7372ea174eb456894f8d2b5

**Comment by tml1024** [comment]
As this canvas thing is being fairly completely re-written in the master branch anyway, it is possible that fixing it in the co-6-4 branch is not worth it.

**Comment by mmeeks** [comment]
Interesting; probably when Gokay's initial work is merged we should re-visit default zoom levels, and how we present them to users.

Possibly we should factor a device-scale out of our presented zoom factor, or allow a larger range of zooms, and pick a better / synthetic '100%' for people ;-) but currently we're trying to get a number of silly things out of the code first before re-complicating the selection of default zoom factor. So I'd shelve this one for the next week or two, but it belongs in that big ball of twine for you Gokay =)

**Comment by gokaysatir** [comment]
Hello :)
@accolon thank you very much for your feedback, we are working on these features and i will inform you as soon as a news arrives:)

**Comment by AZweimiller** [comment]
This behavior is also visible in Windows 10 with DPI scaling enabled. I have the system-wide DPI scaling at 200% and I have to manually zoom each Sheet to 200% for it to look normal.

**Comment by celogeek** [comment]
About retina display, the pixel density is twice the normal one.
I have found this function

```
function isRetinaDisplay() {
        if (window.matchMedia) {
            var mq = window.matchMedia("only screen and (min--moz-device-pixel-ratio: 1.3), only screen and (-o-min-device-pixel-ratio: 2.6/2), only screen and (-webkit-min-device-pixel-ratio: 1.3), only screen  and (min-device-pixel-ratio: 1.3), only screen and (min-resolution: 1.3dppx)");
            return (mq && mq.matches || (window.devicePixelRatio > 1)); 
        }
    }
```

I try it, with retina resolution, it return true, and without it return false.

is it possible to integrate this function to adapt the size of the cell? This way we don't need to zoom in each time we open a spreadsheet.

Alternatively, is they a way to set up the default zoom ?

**Comment by mmeeks** [comment]
There is a big re-write of how we render to get great crispness at various different zoom levels that we are testing in online master currently. If you would like to build and test that it would be very helpful.
We hope to start back-porting this to cp-6.4 next week - and (I hope) we can address this issue as part of that. @celogeek - you're quite right, we should be picking a sensible zoom based on the devicePixelRatio - or scaling what our UX shows as the zoom level to match that (or some other such trick). The re-write should also ensure that browser zoom can be used safely independently of document zoom to scale the UX elements.

**Comment by thebearon** [comment]
This is fixed in 6.4.7.

**Comment by celogeek** [comment]
I just test it. Wonderful improvement! The interface looks way much cleaner!!! Thanks.
